### PR TITLE
Move all options to global file

### PR DESCRIPTION
**Pull request by lukekoko**
Further work on moving options to global options file `Options.h/.cpp`. Based on [Pull #373](https://github.com/SVF-tools/SVF/pull/373)

### CODE REVIEW

**Comment by yuleisui** [comment]
It looks good to me apart from the naming conventions. Can we use/fix the namings of the variables to be consistent (e.g., uppercase or lowercase for the first alphabet of each name). @mbarbar would you like to take a look and any comments?

**Comment by mbarbar** [comment]
A brief look and it looks good to me, good job.

With respect to naming, I think most important would be the option names (`./bin/wpa -help`). Currently we have a few conventions, some examples: `fieldlimit` vs. `dump-icfg` vs. `dump-VFG` vs. `optSVFG` vs. ...

These are user-facing and should probably all be the same. My personal vote goes to `each-word-abc`, so all lowercase and hyphenated.

With respect to naming of the internal variables (i.e., the `cl::opt` objects), they are also inconsistent. It's up to you @yuleisui if/how you want them fixed. There are a fair few bits of style inconsistencies in SVF at the moment (lots of people working on it, too many deadlines, whatever). Maybe making `clang-format` (+ choosing or making a style) part of CI could solve this problem permanently, I think you mentioned added a style checker a while back.

Finally, we should probably stick with `Class::field` for `::` as everywhere in SVF uses this (except for something I added a few years back, it appears........). I think running the following, then inspecting the diff for anything weird should work:
```
grep -Rl 'Options :: ' lib/ include/ | xargs sed -i 's/Options :: /Options::/g'
```

**Comment by yuleisui** [comment]
> A brief look and it looks good to me, good job.
> 
> With respect to naming, I think most important would be the option names (`./bin/wpa -help`). Currently we have a few conventions, some examples: `fieldlimit` vs. `dump-icfg` vs. `dump-VFG` vs. `optSVFG` vs. ...
> 


> These are user-facing and should probably all be the same. My personal vote goes to `each-word-abc`, so all lowercase and hyphenated.

> 
> With respect to naming of the internal variables (i.e., the `cl::opt` objects), they are also inconsistent. It's up to you @yuleisui if/how you want them fixed. There are a fair few bits of style inconsistencies in SVF at the moment (lots of people working on it, too many deadlines, whatever). Maybe making `clang-format` (+ choosing or making a style) part of CI could solve this problem permanently, I think you mentioned added a style checker a while back.
> 
> Finally, we should probably stick with `Class::field` for `::` as everywhere in SVF uses this (except for something I added a few years back, it appears........). I think running the following, then inspecting the diff for anything weird should work:
> 
> ```
> grep -Rl 'Options :: ' lib/ include/ | xargs sed -i 's/Options :: /Options::/g'
> ```

**Comment by yuleisui** [comment]
> A brief look and it looks good to me, good job.
> 
> With respect to naming, I think most important would be the option names (`./bin/wpa -help`). Currently we have a few conventions, some examples: `fieldlimit` vs. `dump-icfg` vs. `dump-VFG` vs. `optSVFG` vs. ...
> 
> These are user-facing and should probably all be the same. My personal vote goes to `each-word-abc`, so all lowercase and hyphenated.
> 

Yes agree. We also need to make the actual command line option consistent. 

> With respect to naming of the internal variables (i.e., the `cl::opt` objects), they are also inconsistent. It's up to you @yuleisui if/how you want them fixed. There are a fair few bits of style inconsistencies in SVF at the moment (lots of people working on it, too many deadlines, whatever). Maybe making `clang-format` (+ choosing or making a style) part of CI could solve this problem permanently, I think you mentioned added a style checker a while back.

Yes, I will add this in the list. @lukekoko maybe can help with this.
> 
> Finally, we should probably stick with `Class::field` for `::` as everywhere in SVF uses this (except for something I added a few years back, it appears........). I think running the following, then inspecting the diff for anything weird should work:
> 
> ```
> grep -Rl 'Options :: ' lib/ include/ | xargs sed -i 's/Options :: /Options::/g'
> ```
Good, please have a look @lukekoko  thanks.

**Comment by lukekoko** [comment]
Need to update test-suite with new option names.
[Pull #35 Test-Suite](https://github.com/SVF-tools/Test-Suite/pull/35)

File: lib/WPA/WPAPass.cpp [hunk 1]
```diff
@@ -81,12 +82,6 @@ static llvm::cl::bits<WPAPass::AliasCheckRule> AliasRule(llvm::cl::desc("Select
             clEnumValN(WPAPass::Veto, "veto", "return NoAlias if any pta says no alias")
         ));
 
-static llvm::cl::opt<bool> anderSVFG("svfg", llvm::cl::init(false),
```
Comment (yuleisui) [inline lib/WPA/WPAPass.cpp:84]: static llvm::cl::bits<PointerAnalysis::PTATY> PASelected(llvm::cl::desc("Select pointer analysis"),
        llvm::cl::values(
            clEnumValN(PointerAnalysis::Andersen_WPA, "nander", "Standard inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenLCD_WPA, "lander", "Lazy cycle detection inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenHCD_WPA, "hander", "Hybrid cycle detection inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenHLCD_WPA, "hlander", "Hybrid lazy cycle detection inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenSCD_WPA, "sander", "Selective cycle detection inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenSFR_WPA, "sfrander", "Stride-based field representation includion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenWaveDiff_WPA, "wander", "Wave propagation inclusion-based analysis"),
            clEnumValN(PointerAnalysis::AndersenWaveDiff_WPA, "ander", "Diff wave propagation inclusion-based analysis"),
            clEnumValN(PointerAnalysis::Steensgaard_WPA, "steens", "Steensgaard's pointer analysis"),
            // Disabled till further work is done.
            // clEnumValN(PointerAnalysis::AndersenWaveDiffWithType_WPA, "andertype", "Diff wave propagation with type inclusion-based analysis"),
            clEnumValN(PointerAnalysis::FSSPARSE_WPA, "fspta", "Sparse flow sensitive pointer analysis"),
            clEnumValN(PointerAnalysis::FSTBHC_WPA, "fstbhc", "Sparse flow-sensitive type-based heap cloning pointer analysis"),
            clEnumValN(PointerAnalysis::VFS_WPA, "vfspta", "Versioned sparse flow-sensitive points-to analysis"),
            clEnumValN(PointerAnalysis::TypeCPP_WPA, "type", "Type-based fast analysis for Callgraph, PAG and CHA")
        ));
static llvm::cl::bits<WPAPass::AliasCheckRule> AliasRule(llvm::cl::desc("Select alias check rule"),
        llvm::cl::values(
            clEnumValN(WPAPass::Conservative, "conservative", "return MayAlias if any pta says alias"),
            clEnumValN(WPAPass::Veto, "veto", "return NoAlias if any pta says no alias")
        ));

These should also be moved to options.cpp. Please keep the same names.

File: lib/Util/Options.cpp [hunk 2]
```diff
@@ -1,33 +1,628 @@
...
+    const llvm::cl::opt<bool> Options::PrintPathCond(
+        "print-pc", 
+        llvm::cl::init(false),
+        llvm::cl::desc("Print out path condition")
+    );
+
+
+    // SVFUtil.cpp
+    const llvm::cl::opt<bool> Options::DisableWarn(
+        "d-warn",
```
Comment (yuleisui) [inline lib/Util/Options.cpp:541]: dward, should be good in this case. please change it back.

File: lib/Util/Options.cpp [hunk 3]
```diff
@@ -1,33 +1,628 @@
...
+    const llvm::cl::opt<bool> Options::DisableWarn(
+        "d-warn", 
+        llvm::cl::init(true),
+        llvm::cl::desc("Disable warning")
+    );
+
+    
+    // Andersen.cpp
+    const llvm::cl::opt<bool> Options::ConsCGDotGraph(
+        "dump-cons-g",
```
Comment (yuleisui) [inline lib/Util/Options.cpp:549]: dump-cons-g ==> dump-constraint-graph

File: lib/Util/Options.cpp [hunk 4]
```diff
@@ -1,33 +1,628 @@
...
+    
+    // Andersen.cpp
+    const llvm::cl::opt<bool> Options::ConsCGDotGraph(
+        "dump-cons-g", 
+        llvm::cl::init(false),
+        llvm::cl::desc("Dump dot graph of Constraint Graph")
+    );
+
+    const llvm::cl::opt<bool> Options::PrintCGGraph(
+        "print-cons-g",
```
Comment (yuleisui) [inline lib/Util/Options.cpp:555]: print-cons-g => print-constraint-graph

File: lib/DDA/DDAPass.cpp [hunk 5]
```diff
@@ -20,30 +21,6 @@ using namespace SVFUtil;
...
-
-static llvm::cl::opt<bool> printCPts("cpts", llvm::cl::init(false),
-                                     llvm::cl::desc("Dump conditional points-to set "));
-
-static llvm::cl::opt<bool> printQueryPts("print-query-pts", llvm::cl::init(false),
-        llvm::cl::desc("Dump queries' conditional points-to set "));
-
-static llvm::cl::opt<bool> WPANUM("wpanum", llvm::cl::init(false),
-                                  llvm::cl::desc("collect WPA FS number only "));
-
```
Comment (yuleisui) [inline lib/DDA/DDAPass.cpp:46]: /// register this into alias analysis group
//static RegisterAnalysisGroup<AliasAnalysis> AA_GROUP(DDAPA);
static llvm::cl::bits<PointerAnalysis::PTATY> DDASelected(llvm::cl::desc("Select pointer analysis"),
        llvm::cl::values(
            clEnumValN(PointerAnalysis::FlowS_DDA, "dfs", "Demand-driven flow sensitive analysis"),
            clEnumValN(PointerAnalysis::Cxt_DDA, "cxt", "Demand-driven context- flow- sensitive analysis")
        ));


The above should also be moved to options.cpp

File: lib/Util/Options.cpp (hunk 2 above)
Comment (yuleisui) [reply lib/Util/Options.cpp:541]: dwarn

**Pull request merged**